WIP: Muon Importer fixes

bsclifton opened the pull request:
Fixes brave/brave-browser#2421
Fixes brave/brave-browser#2415

(more to come- see https://github.com/brave/brave-browser/projects/12)

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
TBD

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

bsclifton commented on browser/importer/brave_profile_writer.cc:
The above logic is a little fragile (could break if we switch our defaults); There isn't an existing way to get the "private search" ID that I could find

bsclifton replied:
Fixed with https://github.com/brave/brave-core/pull/1064/commits/d3d4a41f9ba001a36524feb4496464b794c4af8e

mkarolin commented on browser/importer/brave_profile_writer.cc:
```diff
@@ -441,3 +447,72 @@ void BraveProfileWriter::UpdateWindows(
+
+// NOTE: the strings used as keys match the values found in Muon:
+// browser-laptop/js/data/searchProviders.js
+std::map<std::string,TemplateURLPrepopulateData::PrepopulatedEngine>
```
Just nitpicks, seems like this could be: static const std::map<std::string, const TemplateURLPrepopulateData::PrepopulatedEngine>

mkarolin commented on browser/importer/brave_profile_writer.cc:
const, as well.

mkarolin commented on browser/importer/brave_profile_writer.cc:
const, as well.

darkdh commented on utility/importer/brave_importer.cc:
```diff
@@ -122,6 +122,13 @@ void BraveImporter::StartImport(const importer::SourceProfile& source_profile,
+// These import types don't need a distinct checkbox in the import screen.
+void BraveImporter::ImportRequiredItems() {
+  ImportReferral();
+  ImportSettings();
```
why don't you use importer::SEARCH_ENGINES for search engine importing?

bsclifton replied:
I think it should always be imported; but we could make a UI flag for it. I asked @rebron @davidtemkin and @bbondy on Slack (waiting for response)

darkdh replied:
it is different between always be imported and checkbox with default checked.
Both achieve the same goal but the later one allows user to opt out

bsclifton replied:
This should be good for now - per convo in Slack
https://bravesoftware.slack.com/archives/C70UTN17S/p1544637587376900

darkdh commented on browser/importer/brave_profile_writer.cc:
can you leverage `ProfileWriter::AddKeywords` for importing regular profile search engine?

bsclifton replied:
I don't think we can use that; it seems that's for adding new engines altogether, versus choosing an already existing one. It also doesn't seem to set a default

bsclifton commented:
Reduced scope of this PR, removed WIP label. Originally, I was going to put multiple fixes in here, but I think just these two are a great start.

I added a test plan and ran into issues when manually testing it. Basically, the private tab settings didn't import properly. Looking at this now...

bsclifton commented:
OK private windows now properly respect the DDG true/false value

I found a problem with Tor windows that I think is outside of the scope of this change. This PR will properly set `kUseAlternativeSearchEngineProvider` (which gets inherited by guest window). However, the engine doesn't seem to respect that value:
https://github.com/brave/brave-core/blob/master/browser/search_engines/tor_window_search_engine_provider_service.cc#L51-L67

bsclifton commented:
@mkarolin @darkdh ready for re-review 😄 👍

darkdh commented:
lgtm, please clean up commits then I will approve

bsclifton commented:
@darkdh commits squashed! 😄👌

kjozwiak commented:
@bsclifton uplift request to `0.58.x` approved after deliberating with @srirambv & @rebron 👍 Please ensure that the correct labels are added/removed and the associated issue is moved to the correct milestone.

bsclifton commented:
0.58.x broke after the above cherry-pick; fixed with https://github.com/brave/brave-core/pull/1090
